src/main/kotlin/net/rrerr/minetrix/MatrixRoomMessageReceivedListener.kt

```diff
@@ -82,4 +87,15 @@ class MatrixRoomMessageReceivedListener(private val plugin: Main) : RoomMessageR
 
     private fun getMediaUrl(message: UrlMessageContent) =
         "${plugin.matrixClient!!.hs.base}/_matrix/media/v1/download/${message.url!!.toString().replace("mxc://", "")}"
+
+    @Throws(NoTokenException::class)
+    private fun sendReadMarker(event: RoomMessageReceivedEvent) {
+        val eventId = event.id
+        val id = plugin.room!!.id
+        val parser = Parser.default()
+        val body = parser.parse(StringBuilder("{\"m.fully_read\": \"$eventId\"}")) as JsonObject
```

Comment:
Also, please use `JsonObject()` directly instead of generating and parsing a string.  It's much safer and more nice

Reply:
I completely missed that in the docs while writing this